src/extensions/agent_modules/shared/state_store/src/agent_state_store.c

```diff
@@ -310,8 +310,13 @@ ADUC_STATE_STORE_RESULT ADUC_StateStore_SetTopicSubscribedStatus(const char* top
 
     if (subscribed)
     {
-        *state_topic_target = (char*)calloc(topic_len + 1, sizeof(char));
-        if (!ADUC_Safe_StrCopyN(*state_topic_target, topic, topic_len + 1, topic_len))
+        const size_t TopicByteLenPlusNullTerm = topic_len + 1;
+        *state_topic_target = (char*)calloc(TopicByteLenPlusNullTerm, sizeof(char));
+        if (!ADUC_Safe_StrCopyN(
+                *state_topic_target /* dest */,
+                TopicByteLenPlusNullTerm /* destByteLen */,
+                topic /* src */,
+                topic_len /* srcByteLen */))
```

Comment:
see my comment in another PR you have out.  isn't src expected to be null terminated? If so, why are you passing srcByteLen to this method?

Reply:
@JeffMill src is not necessarily null-terminated and caller could want substring.